### ISSUE TITLE
Casting Table output name offset

### ISSUE THREAD

**Issue by Shahelion** [ReikaKalseki/Reika_Mods_Issues#554]
The mouse is currently over the compass in the output slot

![image](https://cloud.githubusercontent.com/assets/6917535/12212983/d60a2d70-b625-11e5-8ed1-f46be369b2a3.png)

**Comment by ReikaKalseki** [comment]
It is a function of GUI scale. It works perfectly on auto. And I cannot get it working on anything else.

What is more, it is working correctly for determining "is mouse over slot", but not the render position.

**Comment by Shahelion** [comment]
This IS on Auto scale. And slightly relevant, why can't I make the compass? The book doesn't say it requires anything else.

**Comment by ReikaKalseki** [comment]
It is working fine for me on Auto, and I could only reproduce your issue on Gui scale normal.

As for the recipe, you lack progression.

**Comment by Shahelion** [comment]
Oh ok woops it was on Normal, Auto looks the same, I just forgot I'd changed it a while ago.

**Comment by ljfa-ag** [comment]
You can take the GUI scale factor into account using the ScaledResolution class.

**Comment by ReikaKalseki** [comment]
I know, and that did not help align the tooltip, as 1) the mouse position code is working for the "is mouse at X" test, meaning adding the GUI scale to the mouse code would break it, and 2) no simple scaling factor fixes the issue.

**Comment by TomeWyrm** [comment]
> And slightly relevant, why can't I make the compass? The book doesn't say it requires anything else.

As Reika said, you lack progress... the fragments have a very obnoxious habit of letting you learn the recipe for things you cannot craft.

_hint_ _hint_ You might want to lock discovery of crafting fragments behind ALL the progress required to craft them.

Also, make certain you're checking EVERY tab of the recipe... once you get past the table stage, there tends to be some things you need to change with the structure in order to craft things... which are indicated, but it took me way too long to notice :-p

**Comment by Barhandar** [comment]
Rock Melter does the same thing with its fluid tank ooltip. Strangely, most other things with fluids (like engines or Reservoirs) in RoC don't.

**Comment by Shahelion** [comment]
Does anyone else have this issue still?

**Comment by ReikaKalseki** [comment]
The relevant code has not been changed.

**Comment by Shibva** [comment]
you know, I think it may be because of the method done to render them

it might not be happening to reika as much because of their screen. I have this issue and I have a screen resolution of 1680 x 1050

it could be possible that its not staing fully relative due to window size; I've seen this happen once with some mod when I adjusted my window  into an unusual shape

**Comment by Shibva** [comment]
I decided to try one thing and I found my theory is correct

here is the thing broke on the UI's ( a good number of them)
![image](https://user-images.githubusercontent.com/38770780/180707276-72ca9ede-61bb-40fe-8ef3-a4aa72e2c05f.png)

and here is the same image with the background; but adjusted the window size of my game, and the issue is fixed magically fixed
![image](https://user-images.githubusercontent.com/38770780/180707613-b8406a50-60ef-4403-a51b-c2a0d322c4a0.png)

in conclusion @ReikaKalseki  @Zsashas and @TomeWyrm @Barhandar  and @ljfa-ag the issue is the result of (what Reika would label as) "dumb code" due to it not being made to adapt to screen resolutions (as the window resizing has demonstrated

maybe with this observation this long-standing issue finally be patched up; with it being clear (hopefully) on what adjustments need to be done to the Ui

hope this helps ^w^

**Comment by Shibva** [comment]
futher shots (NEI adaptation to a JEI esque format goes to GTNH devs)

shows that its working normally on their renders and time renders
![image](https://user-images.githubusercontent.com/38770780/180709286-5e860e5b-2eb4-40d1-9eef-0426d2e9a5d6.png)

this proves that its exclusively tooltips popup not properly taking into account resolution sizes (or what the proper term is, ratios? idk)
![image](https://user-images.githubusercontent.com/38770780/180709450-ea4ce835-de6c-4d70-885a-ff3d899b9721.png)
